sklearn/impute/tests/test_knn.py

```diff
@@ -598,8 +601,10 @@ def test_knn_imputer_drops_all_nan_features(na):
     assert_allclose(knn.transform(X2), X2_expected)
 
 
+@pytest.mark.parametrize("working_memory", [None, 0])
```

Comment:
Wouldn't setting `working_memory` to 0 trigger a UserWarning?

Reply:
Where does this raise a UserWarning? Do you mean that you think it should. You're right that elsewhere we have tested with eps, rather than 0, but I think 0 is okay... It just forces the alg to take the minimum chunk possible.

Reply:
Oh, I see. Yes, it should raise a warning in get_chunk_n_rows.

Reply:
Not sure how to test this easily but avoid warning.

Reply:
we can maybe decorate the test and filter the warnings since we are expecting it?

Reply:
I think it is strange we are testing something that raises a warning. Are we going to expect users to set `working_memory` to 0 which in turns sets `chuck_n_rows` to one? If so, we can filter the warning here.

Reply:
I might have misunderstood the test but I think we set it to 0 because we work with a really small array (and we still want to test the chunking). In practice, people will have a huge array and we will work with a couple of MB (so not 0).

Reply:
I did not think of that! Then we should leave a comment regarding why we setting `working_memory=0` and filter the warning.

Reply:
Oh yes. Without this we don't test the chunking.

Reply:
+1 for the small comment